tests/util/test_statistic_threshold.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+
+import itertools
+import math
+import numbers
+import random
+import numpy.random
+
+SMYTHE_THOMPSON_NUMBER = 17
+
+
+def nones(n):
+    "Returns an iterator which generates None n times"
+    return itertools.repeat(None, n)
+
+
+def lbeta(m, n):
+    """Return log(Beta(m,n))"""
+    return math.lgamma(m) + math.lgamma(n) - math.lgamma(m + n)
+
+
+def failprob_threshold(observed, ns, threshold):
+    """Takes ([float] observed, int ns, float threshold)
+
+    observed: iid numeric samples of some test statistic,
+    ns: number of iid samples to be drawn when the test statistic is being used
+        in an integration test,
+    threshold: desired maximum probability that "ns" iid samples from
+    the same underlying distribution "observed" was drawn from are all less
+    than some bound "x".
+
+    Returns (float p, float x). Return value "p" is the posterior probability
+    of drawing "ns" samples from the underlying distribution which are all less
+    than "x" is less than "threshold". "x" is the ML estimate of the
+    threshold**(1/ns) quantile"""
+
+    # Type checking
+    if not all(isinstance(d, numbers.Number) for d in observed):
+        raise ValueError, 'observed is not a list of numeric values'
+    if (threshold > 1) or (threshold < 0):
+        raise ValueError, 'threshold is not a probability'
+    if (round(ns) != ns) or (ns < 1):
+        raise ValueError, 'ns is not a natural number'
```

Comment:
FYI: This syntax is deprecated and won't work in Python 3. Instead:

    raise ValueError('ns is not a natural number')